app/supplejack/extraction/execution.rb

```diff
@@ -74,7 +98,30 @@ def custom_stop_conditions_met?
       stop_conditions = @extraction_definition.stop_conditions
       return false if stop_conditions.empty?
 
-      stop_conditions.map { |condition| condition.evaluate(@de.document.body) }.any?(true)
+      triggered_conditions = stop_conditions.select { |condition| condition.evaluate(@de.document.body) }
```

Comment:
You might be able to put this in the evaluation of the stop condition rather than here, so you can evaluate like normal and if the block returns true you can log that it was hit